pressbooks/pressbooks-multi-institution
# feat: add institution column to book list

Issue #40. Requires the following PR https://github.com/pressbooks/pressbooks-network-analytics/pull/331.

This PR hooks into the newly created filters in network analytics to extend the behaviour of the book list. It includes anew `institution` column and adds a new filter to help users refine their searches. It will also include the new column in the CSV download feature.

### Testing case

As a Super Admin or Network Manager:

- Visit the Book List page
- There should be a new `institution` column displayed containing the name of the institution the book is assigned to
- This column should be sortable (unassigned books will be displayed first in ascending order)
- Filters should work as before with the addition of the new `Institution` filter tab
- CSV file should work as before and should include the new column

As an Institutional Manager

- Visit the Book List page
- The page title should display the name of the institution you're assigned to
- Total books and storage should reflect only the books assigned to that particular institution
- Only books assigned to the same institution should be visible
- The institution column should be displayed as well but with no value other than the user's institution
- You should not see the new `Institution` filter tab
- CSV download should include the `Institution` column

> [!NOTE]
> When this plugin is deactivated, the book list should continue to work as before
